structure-message/samples/generated-sample.json

```diff
@@ -0,0 +1,4303 @@
+										"rel": "a",
+										"urn": "a",
+										"uri": "http://www.example.org/data.json",
+										"title": "a",
+										"titles": {
+											"en-GB-oed": "a"
+										},
+										"type": "a",
+										"hreflang": "en-GB-oed"
+									}
+								],
+								"localID": "$",
+								"condition": "a",
+								"conditions": {
+									"en-GB-oed": "a"
+								},
+								"targetStep": "a"
+							}
+						]
+					}
+				]
+			}
+		],
+		"categorisations": [
+			{
+				"id": "$",
+				"version": "0.0",
+				"agencyID": "A.A",
+				"isExternalReference": true,
+				"isFinal": true,
+				"validFrom": "2017-11-16T11:04:00Z",
+				"validTo": "2017-11-16T11:04:00Z",
+				"name": "a",
+				"names": {
+					"en-GB-oed": "a"
+				},
+				"description": "a",
+				"descriptions": {
+					"en-GB-oed": "a"
+				},
+				"annotations": [
+					{
+						"id": "a",
+						"title": "a",
+						"type": "a",
+						"text": "a",
+						"texts": {
+							"en-GB-oed": "a"
+						},
+						"links": [
+							{
+								"href": "http://www.example.org/data.json",
+								"rel": "a",
+								"urn": "a",
+								"uri": "http://www.example.org/data.json",
+								"title": "a",
+								"titles": {
+									"en-GB-oed": "a"
+								},
+								"type": "a",
+								"hreflang": "en-GB-oed"
+							}
+						]
+					}
+				],
+				"links": [
+					{
+						"href": "http://www.example.org/data.json",
+						"rel": "a",
+						"urn": "a",
+						"uri": "http://www.example.org/data.json",
+						"title": "a",
+						"titles": {
+							"en-GB-oed": "a"
+						},
+						"type": "a",
+						"hreflang": "en-GB-oed"
+					}
+				],
+				"source": "a",
+				"target": "a"
+			}
+		],
+		"contentConstraints": [
+			{
+				"type": "Allowed",
+				"id": "$",
+				"version": "0.0",
+				"agencyID": "A.A",
+				"isExternalReference": true,
+				"isFinal": true,
+				"validFrom": "2017-11-16T11:04:00Z",
+				"validTo": "2017-11-16T11:04:00Z",
+				"name": "a",
+				"names": {
+					"en-GB-oed": "a"
+				},
+				"description": "a",
+				"descriptions": {
+					"en-GB-oed": "a"
+				},
+				"annotations": [
+					{
+						"id": "a",
+						"title": "a",
+						"type": "a",
+						"text": "a",
+						"texts": {
+							"en-GB-oed": "a"
+						},
+						"links": [
+							{
+								"href": "http://www.example.org/data.json",
+								"rel": "a",
+								"urn": "a",
+								"uri": "http://www.example.org/data.json",
+								"title": "a",
+								"titles": {
+									"en-GB-oed": "a"
+								},
+								"type": "a",
+								"hreflang": "en-GB-oed"
+							}
+						]
+					}
+				],
+				"links": [
+					{
+						"href": "http://www.example.org/data.json",
+						"rel": "a",
+						"urn": "a",
+						"uri": "http://www.example.org/data.json",
+						"title": "a",
+						"titles": {
+							"en-GB-oed": "a"
+						},
+						"type": "a",
+						"hreflang": "en-GB-oed"
+					}
+				],
+				"type": "Allowed",
+				"constraintAttachment": {
```

Comment:
In SDMX-ML it is not allowed to have different types of Constraint Attachments within the same Constraint.
I understand that this is an autogenerated message. Is there a way to pose the above restriction?

Reply:
Removed wrong duplicate of type. Also corrected uri/urn paths as JSON validators complained.